src/gui_writebox.cpp

```diff
@@ -0,0 +1,64 @@
+#include "gui_writebox.hpp"
+
+Writebox::Writebox(float arg_x, float arg_y, int arg_character_limit) :
+	box{arg_x, arg_y, width, height},
+	box_hitbox{arg_x - width * 0.5f, arg_y - height * 0.5f, width, height},
+	character_limit(arg_character_limit) {}
+
+bool Writebox::is_active() const
+{
+	return active;
+}
+
+void Writebox::set_active(bool arg)
+{
+	active = arg;
+}
+
+void Writebox::set_position(float x, float y)
+{
+	box.x = x;
+	box.y = y;
+	box_hitbox.x = x - box.width * 0.5f;
+	box_hitbox.y = y - box.height * 0.5f;
+}
+
+void Writebox::handle_events(int key_pressed)
+{
+	if (IsMouseButtonPressed(MOUSE_BUTTON_LEFT))
+		active = CheckCollisionPointRec(GetMousePosition(), box_hitbox);
+
+	if (active)
+	{
+		if (IsKeyPressed(KEY_BACKSPACE) && !text.empty())
+		{
+			text.pop_back();
+			if (prefix > 0)
+				--prefix;
+		}
+		else if (key_pressed > 0 && text.size() < character_limit)
+			text.push_back(static_cast<char>(key_pressed));
+	}
+}
+
+void Writebox::draw() const

```

Comment:
Ta cała funkcja bardzo mi się nie podoba. Poza nie do końca użyciem std::string_view, krokowaniem o 1 zamiast robienie czegoś a.la. binary search, oraz ignorowaniem potencjalnych ligatur (nie wiem, czy raylib je obsługuje), nie jestem do końca pewny, jaki jest cel tej pętli. Czy chodzi o wypisanie tylko końca tekstu?

Reply:
Chodzi o to, zeby tekst nie wychodzil poza granice boxa, gdy tekst jest wydluzany to ta petla sobie zwiekszy prefix w miare potrzeby (bo z roznymi znakami roznie to bywa, np. "i" jest tak waskie, ze nie przy kazdym trzeba zwiekszyc prefix, a na niektorych ukladach jak dodasz sobie "m" to trzeba zwiekszyc prefix o wiecej niz 1, wiec no xD). Z usuwaniem znakow tez sie pare rzeczy dzieje, ktore ta petla pozniej automatycznie oczyszcza, troche to wszystko clunky, ja wiem, ale ostateczny efekt mnie zadowalal, wiec stwierdzilem, ze niech tak juz zostanie